lib/TTT/validity.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Validity do
+  def user_move_to_internal_state(move) do
+    try do
+      Map.get(
+        %{
+          1 => {0, 0},
+          2 => {0, 1},
+          3 => {0, 2},
+          4 => {1, 0},
+          5 => {1, 1},
+          6 => {1, 2},
+          7 => {2, 0},
+          8 => {2, 1},
+          9 => {2, 2}
+        },
+        String.to_integer(move),
```

Comment:
This coercion from string to integer could possibly happen in your `input` module, or maybe before the move gets passed to validity.  Validity could potentially be better off in a world where it can trust the types of the things being handed to it.